### PR TITLE
[match] Added basic authorization to git storage

### PR DESCRIPTION
**Pull request by bguidolim**
### Checklist
- [x] I've run `bundle exec rspec` from the root directory to see all new and existing tests pass
- [x] I've followed the _fastlane_ code style and run `bundle exec rubocop -a` to ensure the code style is valid
- [x] I've read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I've updated the documentation if necessary.

### Motivation and Context
I want to be able to use `match` with git storage on GitHub Actions. Currently, there is no way to set SSH keys, the alternative is using HTTPS with basic authorization.

### Description
A new parameter was added to `match` action to support HTTPS basic authorization, it can be used as environment variable as well. Adding the authorization, `match` will set the git config `http.extraheader`.

### CODE REVIEW

**Comment by janpio** [comment]
Thanks for working on this @bguidolim, sounds very interesting.

Can we maybe add an example of how that would be used somewhere in the docs (https://github.com/fastlane/fastlane/blob/master/fastlane/lib/fastlane/actions/docs/sync_code_signing.md or one of the `match` examples)? That would make this a lot more accessible.

**Comment by bguidolim** [comment]
@janpio done ;)

File: fastlane/lib/fastlane/actions/docs/sync_code_signing.md [hunk 1]
```diff
@@ -92,6 +92,24 @@ app_identifier("tools.fastlane.app")
 username("user@fastlane.tools")
 ```
 
+##### Considerations for GitHub:
```
Comment (janpio) [inline fastlane/lib/fastlane/actions/docs/sync_code_signing.md]:
```suggestion
##### Considerations for GitHub:

```

**Comment by janpio** [comment]
Looks great to me now - left only that someone actually tests it 🚀

**Comment by bguidolim** [comment]
Do I count as a tester? Because I’m using my branch as source and it’s working good with GitHub Actions.

**Review by janpio — APPROVED** [review]
In this case, yes :) Thanks! 🚀

**Comment by kouki-dan** [comment]
I have a question about this PR.
Which token do I set to `MATCH_GIT_BASIC_AUTHORIZATION`?

I tried setting Personal Access Token to this env, but it returns error.
Log is here
```
Cloning into '/var/folders/f6/c2jjnnrj04lgg72scq8kzfsc0000gn/T/d20190927-57277-em7kyi'...
fatal: could not read Username for 'https://github.com': terminal prompts disabled
[13:59:17]: Exit status: 128
[13:59:17]: Error cloning certificates repo, please make sure you have read access to the repository you want to use
[13:59:17]: Run the following command manually to make sure you're properly authenticated:
[13:59:17]: $ git clone https://github.com/kouki-dan/certificates.git /var/folders/f6/c2jjnnrj04lgg72scq8kzfsc0000gn/T/d20190927-57277-em7kyi -c http.extraheader='AUTHORIZATION: basic [Personal Access Token]'
```

The last line explains manually command and I tried it.
I found a token which is encoded by base64 works well.
It works.
```
git clone https://github.com/kouki-dan/certificates.git /var/folders/f6/c2jjnnrj04lgg72scq8kzfsc0000gn/T/d20190927-57277-em7kyi -c http.extraheader='AUTHORIZATION: basic [Base64 Encoded Personal Access Token]'
```

So I set a base64 encoded personal access token as `MATCH_GIT_BASIC_AUTHORIZATION`, but it don't still works because the value is encoded with `shellescape` and base64 encoded string has `=` padding in last of the value. I think `=` is excaped and don't works

Which token do you use to this feature? Not Personal Access Token? 
Are there my mistakes?

Could you tell me usage?

**Comment by janpio** [comment]
Please create a new issue for this and mention the author @bguidolim instead of discussing here.

That being said, this functionality is not rolled out to rubygems/users yet.

**Comment by bguidolim** [comment]
Hi @kouki-dan

For GitHub you should use `Base64.encode('username:personal_token')`. I didn’t want to convert automatically to Base64 because I don’t know how the other providers works. So you should provide the key in the same way you do that in command line.

In you terminal is working probably because your git config has the username set already, in a CI environment you must provide this information.

Maybe what’s missing is more information about how to provide the key to the tool.

Please, do as @janpio said, open a new issue and mention me, so we can discuss if there are any more doubts.

@janpio I still using my branch in production and it’s working fine, I think you can release this feature anyways. Maybe I just to add more documentation or some hints about this.

**Comment by kouki-dan** [comment]
Thank you for your reply and explanation and sorry for asking a question in Pull Request. I will create a new issue when I have a question.

I also tried using base64 encoded `username:personal_token` value to store env and getting same error.
I think when base64 encoded string has a `=` in tail for padding (in case of value is not multiples of four), `shellescape` method escapes `=` as `\=` at [here](https://github.com/fastlane/fastlane/pull/15324/files#diff-d3fbd2e18dbe97ef20b07721c3ae0ba0R72)

I'll try to examine more and create an issue if I know something.

Thank you very much for conversation and fantastic feature!
I really want this feature and I appreciates this PR!

**Comment by bguidolim** [comment]
@kouki-dan I think you're right. I've opened a new PR fixing that: #15391 

Thank you :)

**Comment by kouki-dan** [comment]
Thank you speedy fixing.

I tried to use your repository and branch and it works well :)
Thank you very much!

**Pull request merged**